jwcrypto/jwa.py

```diff
@@ -466,6 +488,991 @@ def algorithm_use(self):
         return 'sig'
 
 
+class _RawKeyMgmt(object):
+
+    def wrap(self, key, keylen, cek, headers):
+        raise NotImplementedError
+
+    def unwrap(self, key, keylen, ek, headers):
+        raise NotImplementedError
+
+
+class _RSA(_RawKeyMgmt):
+
+    def __init__(self, padfn):
+        self.padfn = padfn
+
+    def _check_key(self, key):
+        if not isinstance(key, JWK):
+            raise ValueError('key is not a JWK object')
+        if key.key_type != 'RSA':
+            raise InvalidJWEKeyType('RSA', key.key_type)
+
+    # FIXME: get key size and insure > 2048 bits
+    def wrap(self, key, keylen, cek, headers):
+        self._check_key(key)
+        if not cek:
+            cek = os.urandom(keylen)
+        rk = key.get_op_key('wrapKey')
+        ek = rk.encrypt(cek, self.padfn)
+        return {'cek': cek, 'ek': ek}
+
+    def unwrap(self, key, keylen, ek, headers):
+        self._check_key(key)
+        rk = key.get_op_key('decrypt')
+        cek = rk.decrypt(ek, self.padfn)
+        if len(cek) != keylen:
+            raise InvalidJWEKeyLength(keylen, len(cek))
+        return cek
+
+
+class _Rsa15(_RSA, JWAAlgorithm):
+    def __init__(self):
+        super(_Rsa15, self).__init__(padding.PKCS1v15())
+
+    @property
```

Comment:
Why are you using properties instead of class attributes here? All values are static. If you put them in class vars, then the values can be introspected on the class, too.